File: task2/complex.cpp
//
// Created by Ярослав Ульяненков on 02.10.2021.
//

#include "complex.h"
#include <cstdio>
#include <cmath>


void complex::In(FILE *in) {
    fscanf(in, "%d%d", &x, &y);
}

void complex::InRnd() {
    x = Random();
    y = Random();
}

void complex::Out(FILE *out) {
    fprintf(out, "%s %d %s %d%s %s %f%s", "It's a complex number:", x, "+", y, "*i.", "It's distance:", Decimal(), "\n");
}

double complex::Decimal() {
    return hypot(x, y);
}

File: task3/numbers/complex.py
from . import Number
from math import hypot


class Complex(Number):
    def __init__(self, a: int, b: int):
        super().__init__(a, b)
        self.x = a
        self.y = b

    def decimal(self) -> float:
        return hypot(self.x, self.y)

    def __str__(self) -> str:
        return f'It\'s a complex number: {self.x} + {self.y}*i. It\'s distance: {self.decimal()}.'
